### ISSUE TITLE
What is the state of support for AD of BatchNormalization and DropOut layers?

### ISSUE THREAD

**Issue by tranvansang** [pytorch/functorch#1117]
I have come to this issue from this post.

https://pytorch.org/functorch/stable/notebooks/per_sample_grads.html

## Background

What I am doing requires per-sample gradient (in fact, I migrated from TF, so I do not have much experience with pytorch, but I have a sufficient understanding of NN training).
When reading the post, I could not figure out whether functorch's `vmap` supports AD function of BN and DropOut layers.

In my understanding, these layers are relatively popular.
Because they are not pure functions (different behaviors in training and testing modes), and also not pure (e.g., BN layer accumulates the average across the different forward pass in training mode), which makes me wonder:

## My questions
1. Does functorch's `vmap` support AD function of BN and DropOut layers?
2. If yes, how does it do that?

I tried searching for issues with BatchNormalization or DropOut keywords, but the results were fragmented and I still do not know what is the current state now.
Opacus says that only `EmbeddingBag` is not supported (https://github.com/pytorch/opacus/blob/5aa378ea98df9caf8ca1987ee4d636219267d17e/opacus/grad_sample/functorch.py#L22).
Could anyone tell me the answer?

If possible, updating the docs to clarify the supports for these layers would be great.

Thank you very much.
